Various STM32 flash related testing additions

erwango opened the pull request:
These commits intent to increase flash testing coverage on STM32 socs.
Additionally fix a bug in nvs sample.

Fixes part of STM32 related flash non covered files in #12860.

erwango committed:
samples/subsys/nvs: Add a clean test termination

If nvs sample is run once until max reboot value, the next time it is is
ran on the same board, it will not behave as expected as code will
read reboot counter set to max value and will stop there.
In order to avoid the operation to wipe the flash manually between
2 runs of the test, add a clean termination to the test by resetting
the reboot counter before exiting.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
board: disco_l475_iot1: add storage partition

Add storage partition to the board in order to enable testing on
nvs sample.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
samples/drivers/flash_shell: Add nucleo_f746zg to platform_whitelist

In order to enable build of stm32f7 flash driver in sanitytests,
add nucleo_f746zg to flash_shell platform whitelist.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

b0661 commented on dts/arm/st/f0/stm32f0.dtsi:
```diff
@@ -41,6 +41,7 @@
 				compatible = "soc-nv-flash";
 				label = "FLASH_STM32";
 
+				erase-block-size = <1024>;
```
STM32F0 does not have a uniform page size (eg. STM32F09x does have 2k pages - RM0091 table 5).

erwango replied:
Oups, missed that, txs!

erwango committed:
boards: nucleo_f091rc: Add nvs support

Add nvs support to nucleo_f091rc board. This requires to
add erase-block-size property to stm32f0.dtsi.
Storage partition is set to 4kb at the end of the flash.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

Laczen commented on samples/subsys/nvs/src/main.c:
```diff
@@ -216,6 +216,13 @@ void main(void)
+			break;
 		}
 	}
 }
```
Not a bug fix, but a nice addition. The disadvantage is that it resets the counter and if you reload it will restart the incrementation, before it would just boot and show the max value. I do agree with the change.

Laczen reviewed (approved):
I have no experience with stm32 devices, so my review is limited to the change in the nvs sample.
